RELEASE-NOTES.md

```diff
@@ -1,5 +1,51 @@
 # Prefect Release Notes
 
+## Release 2.14.12
+
+### Dynamic time-span filtering and customizable flow run visualization
```

Comment:
@zhen0 does this heading sound good to you?

Reply:
I think that over plays it a bit. The visualizations haven't changed much.

Reply:
please go ahead and change. it didn't have a title.

Reply:
Maybe "Increased customization of date and time filters across the UI"

Reply:
done!

Reply:
Thank you